lib/libutils/ext/include/util.h

```diff
@@ -70,15 +70,42 @@
 
 /*
  * Round up the even multiple of size and return if result overflow
- * output value range. Size has to be a power of 2.
+ * output value range. @size has to be a power of 2.
  */
-#define ROUNDUP_OVERFLOW(v, size, res) (__extension__({ \
-	typeof(*(res)) __roundup_tmp = 0; \
-	typeof(v) __roundup_mask = (typeof(v))(size) - 1; \
-	\
-	ADD_OVERFLOW((v), __roundup_mask, &__roundup_tmp) ? 1 : \
-		((void)(*(res) = __roundup_tmp & ~__roundup_mask), 0); \
-}))
+ #define __ROUNDUP_OVERFLOW_FAST(v, size, res) \
+	(__extension__({ \
+		typeof(*(res)) __roundup_tmp = 0; \
+		typeof(v) __roundup_mask = (typeof(v))(size) - 1; \
+		\
+		ADD_OVERFLOW((v), __roundup_mask, &__roundup_tmp) ? 1 : \
+			((void)(*(res) = __roundup_tmp & ~__roundup_mask), 0); \
+	}))
+
+#define ROUNDUP_OVERFLOW_FAST(v, size, res) \
+	(__extension__({ \
+		assert(IS_POWER_OF_TWO(size); \
+		__ROUNDUP_OVERFLOW_FAST((v), (size), (res)); \
+	}))
+/*
+ * Round up the even multiple of size and return if result overflow
+ * output value range. @size can have any value.
+ */
+#define __ROUNDUP_OVERFLOW_SLOW(v, size, res) \
+	(__extension__({ \
+		typeof(*(res)) __roundup_tmp = 0; \
+		typeof(v) __roundup_mod = 0; \
+		typeof(v) __roundup_add = 0; \
+		\
+		__roundup_mod = (v) % (typeof(v))(size); \
+		if (__roundup_mod) \
+			__roundup_add = (typeof(v))(size) - __roundup_mod; \
+		ADD_OVERFLOW((v), __roundup_add, &__roundup_tmp) ? 1 : \
+			((void)(*(res) = __roundup_tmp), 0); \
```

Comment:
This implementation is suboptimal. I've tested it successfully through xtest, forcing `ROUNDUP_OVERFLOW()` to use this "slow" path.